### PR TITLE
Add LabelGroup component

### LINKED ISSUE #3829: Labels should be group-able like chip groups and link-able
Chip Groups allow for grouping of chips.  I'd like the same features for labels, that allow group of labels, that also serve as links. 

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
Enhancement

**What is your product and what release version are you targeting?**
Ansible Tower

### PR DESCRIPTION
**Pull request by evwilkin**
**What**: Closes #3829 

This PR adds the `LabelGroup` component, and ads a new `isOverflowLabel` prop to the `Label` component.
Much of the functionality and code was borrowed from the ChipGroup which has similar functionality.

### CODE REVIEW

**Review by mcarrano — DISMISSED** [review]
This looks good @evwilkin with one exception. In the vertical group, looks like the close button is being placed at the bottom and should be in the upper right. This actually might be a core problem and I know we've recently been having some discussion relative to wrapping chip groups. @mcoker I think that the same issues apply here. Must have missed this in the HTML review.

**Comment by mcoker** [comment]
@mcarrano I agree the position of the close button in the vertical group feels awkward, but I talked with Mary about it and mocked up the close button at the top and bottom, and she advised me to put it at the bottom. Do we want to change that?

**Comment by mcarrano** [comment]
OK. Let's discuss then @maryshaks1996 .  My concern is this is something different than what we are doing with the chip group.

**Comment by mcarrano** [comment]
@mcoker I just discussed this with @maryshaks1996 and we agreed to move the close button for the vertical groups to the top-right to be consistent with what we do it Chip groups, alerts, etc. In a sense the vertical group and a horizontal group that wraps would look identical if you made the container narrow enough. Is that a change we can make it core?

**Comment by mcoker** [comment]
@mcarrano yep, would you mind creating an issue for that? Do you want to try and get it in this sprint?

**Comment by mcarrano** [comment]
@mcoker I've opened https://github.com/patternfly/patternfly/issues/3456. Yes, it would be great to get this into the current sprint if possible. Thanks!

**Comment by evwilkin** [comment]
@mcoker @mcarrano if we can get that change in early next week I can update this PR to include that - I'll leave this as-is for now pending that update.

File: packages/react-integration/cypress/integration/labelgroup.spec.ts
Comment (mcoker) [inline packages/react-integration/cypress/integration/labelgroup.spec.ts:1]: there's a bunch of stuff in here labeled 'chip' - should it be 'label'?

**Comment by mcoker** [comment]
@evwilkin LGTM! I think we can merge this when it's done and then make any updates needed for https://github.com/patternfly/patternfly/issues/3456 if necessary since it's still beta. Also I'm not sure if there will be any react update from #3456

**Review by mcoker — DISMISSED** [review]
🏆 👏

File: packages/react-integration/cypress/integration/labelgroup.spec.ts
Comment (evwilkin) [reply packages/react-integration/cypress/integration/labelgroup.spec.ts]: Thanks @mcoker good catch - I've updated the integration tests for this component

**Review by mcoker — DISMISSED** [review]

**Review by jessiehuff — COMMENTED** [review]
For the most part it looks pretty good! 🙂 Keyboard interactions work as expected, and it sounds good in VO. Just one small nit that the `aria-labelledby` in the last example (on the `<ul>`) doesn't actually point to any valid id. It seems that that id is missing? I think it should be on the span. If you run axe, it'll flag this one to point out where I mean.

Also, I see another axe error that I've been debating over with chip group as it exists there as well. It's a little odd to have an `aria-hidden` on the same element that we have a tooltip on, but realistically it doesn't cause any issues here so it's probably fine. That's not specific to your PR, but thought it might be worth noting.

File: packages/react-core/src/components/LabelGroup/examples/LabelGroup.md
Comment (tlabaj) [inline packages/react-core/src/components/LabelGroup/examples/LabelGroup.md:6]: I am not sure why this is not showing as Beta in the preview link.

File: packages/react-integration/demo-app-ts/src/components/demos/LabelGroupDemo/LabelGroupWithCategoryDemo.tsx
Comment (tlabaj) [inline packages/react-integration/demo-app-ts/src/components/demos/LabelGroupDemo/LabelGroupWithCategoryDemo.tsx:1]: Can you add the new `isOverflowLabel` to the demo app.
Can you also make sure all the props for the `LabelGroup` are used at least once in the demo to verify typing.

File: packages/react-core/src/components/LabelGroup/LabelGroup.tsx
Comment (tlabaj) [inline packages/react-core/src/components/LabelGroup/LabelGroup.tsx:1]: So you the `Label` and `LabelGroup` in separate directories here.  @mcarrano  know y had an opinion about the `Chip` and `ChipGroup` being placed in directory  and therefore on the same page for documentation.  Do you feel that same way for the `Label` and `LabelGroup`?

**Review by tlabaj — CHANGES_REQUESTED** [review]

File: packages/react-core/src/components/LabelGroup/examples/LabelGroup.md
Comment (evwilkin) [reply packages/react-core/src/components/LabelGroup/examples/LabelGroup.md:6]: Fixed - corrected from `isBeta` to `beta`

File: packages/react-core/src/components/LabelGroup/LabelGroup.tsx
Comment (evwilkin) [reply packages/react-core/src/components/LabelGroup/LabelGroup.tsx:1]: Yeah I could see it going either way - combining to mirror Chips/ChipGroup, or staying separate as users may have already been using Labels and be thrown off seeing the docs for them disappear (replaced by LabelGroup). @mcarrano happy to defer to your feedback on this, currently they're listed separately.

**Comment by evwilkin** [comment]
@jessiehuff thanks for taking a look! I'm not able to recreate the first issue you mentioned, I am seeing the `<span>` just before the `<ul>` has the correct ID as expected, and axe is not throwing the error for me - are you still seeing this?

Screen Shot 2020-09-11 at 1 34 38 PM

File: packages/react-core/src/components/LabelGroup/LabelGroup.tsx
Comment (mcarrano) [reply packages/react-core/src/components/LabelGroup/LabelGroup.tsx:1]: I could argue either way, but I agree with your reasoning for keeping Label as a thing on it's own. It would be potentially confusing if that went away. I also think that there are many instances when labels will be used by themselves where chips would rarely be used outside of a group. So let's keep separate.

**Review by jessiehuff — DISMISSED** [review]
Oh weird! Maybe the preview hadn't updated, it's not there anymore. LGTM! 🙂

**Review by kmcfaul — APPROVED** [review]

File: packages/react-core/src/components/LabelGroup/LabelGroup.tsx
Comment (evwilkin) [reply packages/react-core/src/components/LabelGroup/LabelGroup.tsx:1]: Thanks!

**Comment by evwilkin** [comment]
@tlabaj updated tests & Beta flag, and per mcarrano above keeping Label separate from LabelGroup as-is. @kmcfaul @jessiehuff if you don't mind re-reviewing, thanks!

**Review by tlabaj — APPROVED** [review]
LGTM

**Review by mcoker — APPROVED** [review]
👍👍

**Review by jessiehuff — APPROVED** [review]

**Review by mcarrano — APPROVED** [review]
I am good with this. @evwilkin so is the understanding then that we will make a core update in the next release then to fix the close button positioning issue?

**Pull request merged**
